### DISCUSSION TITLE
correct way to create desktop app for tracking opcua data

### DISCUSSION THREAD

**Discussion by Deccoyi** [FreeOpcUa/opcua-asyncio#1120] [Q&A] [answered]
Hi. I am trying to create (preferably) a desktop app to get data from a PLC via OPCUA. I've read a lot of documentation about different libraries with different programming languages but I couldn't decide. I've managed to get data with experimeting with examples but combining all into an app with a gui is very complicated.

I thought about using
1. Electron.js + React + Flask (opcu-asyncio)   (but i feel like i am wrong about this. because I couldn't do it yet)
2. Electron.js + React + node-opcua (couldn't do it aswell)

What do you think best parctice would be? I am a beginner btw. Maybe this is so basic of a question  😶

**Comment by oroulet** [comment, marked as answer]
everything will work. it depends what you want to learn

**Comment by AndreasHeine** [comment]
If you want to use Electron and React anyways i would stay in one language to make it simpler!
The application architecture in both cases is similar and from you posted issue its more a design problem.

If you build an App with Electron and node-opcua its quite simple because its all js and in one place (no "backend" all node.js)

**Comment by nkawerau** [comment]
"Hi, I am working on a similar project and I am a beginner as well. My approach is to stay in one Python environment using "opcua-asyncio" for the communication with the server and "Streamlit" for building a web-based data application as an HMI. Since you can use Streamlit offline as well, maybe it suits your use case as well.

How did you proceed and how is it going?"
